src/librustc_mir/interpret/machine.rs

```diff
@@ -210,4 +211,18 @@ pub trait Machine<'a, 'mir, 'tcx>: Sized {
         ecx: &mut InterpretCx<'a, 'mir, 'tcx, Self>,
         extra: Self::FrameExtra,
     ) -> InterpResult<'tcx>;
+
+    fn int_to_ptr(
+        _int: u64,
+        _extra: &Self::MemoryExtra,
+    ) -> InterpResult<'tcx, Pointer<Self::PointerTag>> {
+        Err(InterpErrorInfo::from(InterpError::ReadBytesAsPointer))
```

Comment:
Ah I think this caused the error. Compare this with what `to_ptr` does:
```rust
    pub fn to_ptr(self) -> InterpResult<'tcx, Pointer<Tag>> {
        match self {
            Scalar::Raw { data: 0, .. } => err!(InvalidNullPointerUsage),
            Scalar::Raw { .. } => err!(ReadBytesAsPointer),
            Scalar::Ptr(p) => Ok(p),
        }
    }
```
The default implementation here should raise a different error when the integer is 0.

Reply:
Ok I'm going to change it